src/test/java/frc/robot/util/SWMTest.java

```diff
@@ -82,37 +158,129 @@ public void go(int idx) {
         // set v_ry
         sh.v_ry = test_cases[idx][4];
         */
-        sh.drive.resetOdometry(new Pose2d(
-            test_cases[idx][0],
-            test_cases[idx][1],
+
+        // set the robot position to (x, y) (with zero heading)
+        drive.resetOdometry(new Pose2d(
+            testCase.x,
+            testCase.y,
             new Rotation2d()));
-        sh.drive.driveHeading(test_cases[idx][3], test_cases[idx][4], 0, true);
-        System.err.println("vx expected " + test_cases[idx][3]);
-        System.err.println("vx actual :( " + sh.drive.getChassisSpeeds().vxMetersPerSecond);
 
+        // that should position the robot
+        assertEquals(testCase.x, drive.getPose().getX(), 0.0001);
+        assertEquals(testCase.y, drive.getPose().getY(), 0.0001);
+
+        // set the robot velocities vx, vy and the robot heading
+        drive.driveHeading(testCase.vx, testCase.vy, 0, true);
+        // System.err.println("vx expected " + testCase.vx);
+        // System.err.println("vx actual : " + drive.getChassisSpeeds().vxMetersPerSecond);
+        // make sure the speeds were set
+        assertEquals(testCase.vx, drive.getChassisSpeeds().vxMetersPerSecond, 0.0001);
+        assertEquals(testCase.vy, drive.getChassisSpeeds().vyMetersPerSecond, 0.0001);
+
+        // do the Shoot command calculation
+
+        // initialize will set the speakerPose field from our Alliance
+        sh.initialize();
+        // but we want to force it to be the Blue speaker for now
+        sh.speakerPose = VisionConstants.BLUE_SPEAKER_POSE;
+        assertTrue(VisionConstants.BLUE_SPEAKER_POSE == sh.speakerPose);
+
+        // Execute the Command...
         sh.execute();
+
+        // execute() should set reasonable values for horiz_angle, vert_angle, and exit_vel
+        System.err.printf("execute() returns: va %f, ha %f, and exit_vel %f\n", sh.vert_angle, sh.horiz_angle, sh.exit_vel);
+        // we should not see any NaNs
+        assertFalse(Double.isNaN(sh.vert_angle));
+        assertFalse(Double.isNaN(sh.horiz_angle));
+        assertFalse(Double.isNaN(sh.exit_vel));
+
+        // Shoot assumes v_note = 15 m/s
+        // double v_note = 15.0;
+        // check that equation was followed (we do not know phi_v)
+        // assertEquals(sh.exit_vel, v_note * Math.sin(phi_v) / Math.sin(sh.vert_angle), 0.0001);
```

Comment:
why is this commented out?

Reply:
The testing code does not have access to local variable phi_v; it would have to recreate it. This was also chasing the discrepancy between the three time values; that was resolved by forcing the Blue Alliance and tighteniing the test for selecting the appropriate branch of the quadratic equation. When the Red Alliance was used, the distances were about 16 meters and the branches had a wider spread.